java/pgv-java-stub/src/main/java/io/envoyproxy/pgv/StringValidation.java

```diff
@@ -103,7 +103,7 @@ public static void address(String field, String value) throws ValidationExceptio
     }
 
     public static void hostName(String field, String value) throws ValidationException {
-        if (!CharMatcher.ascii().matchesAllOf(value)) {
+        if (!isAscii(value)) {
```

Comment:
Why is this better than using `CharMatcher`?

Reply:
1. one dependency less
2. no object needs to be created

Reply:
All I said applies to the latest version of the function, I updated it once more to be maximally efficient.

Reply:
Since you are going for maximal efficiency (I agree it is efficient), how is your implementation more efficient than Guava's implementation? From what I can see, you and Google have converged. 😁 

1. Iterate over each character in a string
https://github.com/google/guava/blob/master/guava/src/com/google/common/base/CharMatcher.java#L507-L514
2. Assert each character is <= 127
https://github.com/google/guava/blob/master/guava/src/com/google/common/base/CharMatcher.java#L1283-L1286

Reply:
Just to be very clear here, the difference will be small in many runs but it should definitely be measurable. I can also set up a benchmark if you want. I mainly did the change because it’s so simple.

### Mine
0. `isAscii()` function call
0. `toCharArray()` function call
0. ultra-fast native array copy
0. return
0. array loop (no additional checks)
0. single comparison (_n_ times)
0. (early or final) return

### Guava
0. `CharMatcher.ascii()` function call
0. access static `Ascii.INSTANCE` property
0. create `Ascii` object if not exist (one time hit)
0. return
0. `CharMatcher.machtesAllOf()` function call (abstract)
0. `CharSequence.length()` function call (abstract)
0. return
0. minus one arithmetic (I have no clue why the iterate backwards, a `while` loop would be the most efficient loop on JVM)
0. `CharSequence.charAt()` function call (abstract)
0. index validation 1
0. index validation 2
0. return
0. `CharMatcher.matches()` function call (_n_ times)
0. single comparison
0. return (_n_ times)
0. (early or final) return